cg/apps/sequencing_metrics_parser/parsers/bclconvert_metrics.py

```diff
@@ -0,0 +1,103 @@
+"""This module parses metrics for files generated by the BCLConvert tool using the dragen hardware."""
+import csv
+import logging
+from pathlib import Path
+from typing import Dict, List
+from cg.store.models import SequencingStatistics
+from cg.constants.demultiplexing import SAMPLE_SHEET_DATA_HEADER
+from cg.apps.sequencing_metrics_parser.utils.bclconvert import BclConvertMetrics
+
+LOG = logging.getLogger(__name__)
+
+
+def parse_bcl_convert_metrics_file(
+    bcl_convert_metrics_file_path: Path, sample_sheet_path: Path, quality_metrics_path: Path
+) -> List[BclConvertMetrics]:
+    """Parse the BCLconvert demultiplexing stats file into the BCLconvertMetrics model."""
+    LOG.info(f"Parsing BCLConvert demultiplexing stats file {bcl_convert_metrics_file_path}")
+    demux_metrics: Dict[int, dict] = read_metric_file_to_dict(
+        metric_file_path=bcl_convert_metrics_file_path
+    )
+    quality_metrics: Dict[int, dict] = read_metric_file_to_dict(
+        metric_file_path=quality_metrics_path
+    )
+    sample_sheet: Dict[int, dict] = read_bcl_convert_sample_sheet_file_to_dict(
+        bcl_convert_sample_sheet_path=sample_sheet_path
+    )
+    bcl_convert_model_list: List[BclConvertMetrics] = []
+    for lane in sample_sheet:
+        lane = int(lane)
+        for sample_id in sample_sheet[lane]:
+            bcl_convert_model_list.append(
+                BclConvertMetrics(
+                    sample_internal_id=demux_metrics[lane][sample_id]["SampleID"],
```

Comment:
If we use pydantic models we should be able to get rid of this deep traversing of the container (dict). Perhaps use function on the model to get the data